src/main/java/org/craftercms/engine/store/s3/S3ContentStoreAdapter.java

```diff
@@ -98,16 +104,15 @@ public Context createContext(final String id, final String rootFolderPath, final
                                  final boolean ignoreHiddenFiles, Map<String, String> configurationVariables)
         throws RootFolderNotFoundException, StoreException, AuthenticationException {
 
-        AmazonS3URI uri = new AmazonS3URI(StringUtils.removeEnd(rootFolderPath, DELIMITER));
-
-        ListObjectsV2Request request = new ListObjectsV2Request()
-                                            .withBucketName(uri.getBucket())
-                                            .withPrefix(uri.getKey())
-                                            .withDelimiter(DELIMITER);
-        ListObjectsV2Result result = client.listObjectsV2(request);
+        S3Uri uri = client.utilities().parseUri(URI.create(StringUtils.removeEnd(rootFolderPath, DELIMITER)));
+        ListObjectsV2Request request = ListObjectsV2Request.builder()
+                .bucket(uri.bucket().orElse(""))
```

Comment:
In what scenarios is an empty bucket name a valid configuration? Should we throw an exception here?
Will it fail anyway a few lines below on `listObjectsV2(request)`?

Reply:
The bucket will never be empty. It will throw an exception when parsing S3 URI if the format is incorrect. However, I switched this to `orElseThrow()`, which should never happen.